flang/test/Runtime/no-cpp-dep.c

```diff
@@ -5,6 +5,9 @@ a C compiler.
 
 REQUIRES: c-compiler
 
+MacOS doesn't have a separate libm.
+UNSUPPORTED: system-darwin
```

Comment:
I couldn't make it work with `DEFINE`/`REDEFINE`, because this test needs to produce a binary linked against system's C library. ``%if system-darwin %{-isysroot `xcode-select -p`"/SDKs/MacOSX.sdk"%} %else %{-lm%}`` worked, but it hardcodes part of the sysroot path.

Actually, after digging a bit more, I found out the real issue is that on MacOS a valid `-isysroot` flag is needed for the built clang to find the system headers and libraries. I'm going to update this PR with the needed changes in `flang/test/lit.cfg.py` and `flang/test/lit.site.cfg.py.in`.

Reply:
Thanks, nice investigation! 

Does this mean that this test will fail unless `CMAKE_OSX_SYSROOT` is set?